src/main/java/org/wildfly/security/auth/server/AbstractMechanismAuthenticationFactory.java

```diff
@@ -39,7 +39,7 @@
         this.mechanismConfigurations = mechanismConfigurations;
     }
 
-    SecurityDomain getSecurityDomain() {
+    public SecurityDomain getSecurityDomain() {
```

Comment:
Even if so, I am reasonably sure you'll have to override the method on those classes to make it visible; hence just putting it on the interface which will do that for you.  All implementations of the interface will have the method, so it's probably just best to do that.  It also gives a place where JavaDoc can be written.